AutoTLS UT fails

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
[2021-09-01T02:08:09.469Z] FAIL: tidb_test.go:878: tidbTestSerialSuite.TestTLSAuto

[2021-09-01T02:08:09.469Z] 

[2021-09-01T02:08:09.469Z] tidb_test.go:898:

[2021-09-01T02:08:09.469Z]     c.Assert(err, IsNil)

[2021-09-01T02:08:09.469Z] ... value *errors.withStack = dsn:root@tcp(127.0.0.1:33034)/test?tls=skip-verify: TLS requested but server does not support TLS ("dsn:root@tcp(127.0.0.1:33034)/test?tls=skip-verify: TLS requested but server does not support TLS")
```

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
in ci https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-nightly/detail/tidb-unit-test-nightly/9958/pipeline/127
### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

This looks like a one-time issue where creation of TLS Certificates failed. It looks like the temp storage wasn't created which caused the code creating the certificates to fail on opening `cert.pem` for writing.
```
[2021-09-01T02:07:16.049Z] [2021/09/01 10:07:15.937 +08:00] [WARN] [misc.go:456] ["TLS Certificate creation failed"] [error="open /tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage/cert.pem: no such file or directory"]
[2021-09-01T02:07:16.049Z] [2021/09/01 10:07:15.937 +08:00] [ERROR] [server.go:222] ["secure connection cert/key/ca load fail"] [error="open /tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage/cert.pem: no such file or directory"]
[2021-09-01T02:07:16.049Z] [2021/09/01 10:07:15.937 +08:00] [INFO] [server.go:245] ["server is running MySQL protocol"] [addr=127.0.0.1:0]
[2021-09-01T02:07:16.308Z] 
[2021-09-01T02:07:16.308Z] ----------------------------------------------------------------------
[2021-09-01T02:07:16.308Z] FAIL: tidb_test.go:878: tidbTestSerialSuite.TestTLSAuto
[2021-09-01T02:07:16.308Z] 
[2021-09-01T02:07:16.308Z] tidb_test.go:898:
[2021-09-01T02:07:16.308Z]     c.Assert(err, IsNil)
[2021-09-01T02:07:16.308Z] ... value *errors.withStack = dsn:root@tcp(127.0.0.1:33034)/test?tls=skip-verify: TLS requested but server does not support TLS ("dsn:root@tcp(127.0.0.1:33034)/test?tls=skip-verify: TLS requested but server does not support TLS")
[2021-09-01T02:07:16.308Z] 
```

> This looks like a one-time issue where creation of TLS Certificates failed. It looks like the temp storage wasn't created which caused the code creating the certificates to fail on opening `cert.pem` for writing.
> 
> ```
> [2021-09-01T02:07:16.049Z] [2021/09/01 10:07:15.937 +08:00] [WARN] [misc.go:456] ["TLS Certificate creation failed"] [error="open /tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage/cert.pem: no such file or directory"]
> [2021-09-01T02:07:16.049Z] [2021/09/01 10:07:15.937 +08:00] [ERROR] [server.go:222] ["secure connection cert/key/ca load fail"] [error="open /tmp/1000_tidb/MC4wLjAuMDo0MDAwLzAuMC4wLjA6MTAwODA=/tmp-storage/cert.pem: no such file or directory"]
> [2021-09-01T02:07:16.049Z] [2021/09/01 10:07:15.937 +08:00] [INFO] [server.go:245] ["server is running MySQL protocol"] [addr=127.0.0.1:0]
> [2021-09-01T02:07:16.308Z] 
> [2021-09-01T02:07:16.308Z] ----------------------------------------------------------------------
> [2021-09-01T02:07:16.308Z] FAIL: tidb_test.go:878: tidbTestSerialSuite.TestTLSAuto
> [2021-09-01T02:07:16.308Z] 
> [2021-09-01T02:07:16.308Z] tidb_test.go:898:
> [2021-09-01T02:07:16.308Z]     c.Assert(err, IsNil)
> [2021-09-01T02:07:16.308Z] ... value *errors.withStack = dsn:root@tcp(127.0.0.1:33034)/test?tls=skip-verify: TLS requested but server does not support TLS ("dsn:root@tcp(127.0.0.1:33034)/test?tls=skip-verify: TLS requested but server does not support TLS")
> [2021-09-01T02:07:16.308Z] 
> ```

Seems it fails very frequently https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-nightly/detail/tidb-unit-test-nightly/9962/pipeline/135

--- a/server/tidb_test.go
+++ b/server/tidb_test.go
@@ -886,6 +886,8 @@ func (ts *tidbTestSerialSuite) TestTLSAuto(c *C) {
 	cfg.Status.ReportStatus = false
 	cfg.Security.AutoTLS = true
 	cfg.Security.RSAKeySize = 528 // Reduces unittest runtime
+	err := os.MkdirAll(cfg.TempStoragePath, 0700)
+	c.Assert(err, IsNil)
 	server, err := NewServer(cfg, ts.tidbdrv)
 	c.Assert(err, IsNil)
 	cli.port = getPortFromTCPAddr(server.listener.Addr())

